### PR TITLE
Rename public classes with 'Master' to 'ClusterManager'

### PR DESCRIPTION
**Pull request by tlfeng**
### Description
Replace `master` terminology by `cluster manager` in the public Java APIs to support inclusive language.
The PR deal with the public class name in the repository (except for those covered in issue https://github.com/opensearch-project/OpenSearch/issues/3542).

- Replace `Master` to  `ClusterManager` for all the classes, including all the references to the classes.

The above changes are done by the [Rename refactoring](https://www.jetbrains.com/help/idea/rename-refactorings.html) feature of IntelliJ IDEA, so that both the definition and reference can be renamed.

The next PR will be like https://github.com/opensearch-project/OpenSearch/commit/de21446b0296665c15b0db0b106e1e1be20a08d2, adding back the classes in old name for backwards compatibility.

**List of classes that renamed in this PR:**
`sever` directory:
interface LocalNodeMasterListener -> LocalNodeClusterManagerListener
final class MasterNodeChangePredicate -> ClusterManagerNodeChangePredicate
NotMasterException -> NotClusterManagerException
NoMasterBlockService -> NoClusterManagerBlockService
UnsafeBootstrapMasterCommand - UnsafeBootstrapClusterManagerCommand
MasterService -> UnsafeBootstrapClusterManagerCommand
MasterNotDiscoveredException -> ClusterManagerNotDiscoveredException
RestMasterAction -> RestClusterManagerAction
`test/framework` directory:
FakeThreadPoolMasterService -> FakeThreadPoolClusterManagerService
BlockMasterServiceOnMaster -> BlockClusterManagerServiceOnMaster
BusyMasterServiceDisruption -> BusyClusterManagerServiceDisruption

### Issues Resolved
The first step to resolve issue https://github.com/opensearch-project/OpenSearch/issues/3543
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tlfeng** [comment]
Log 6091 shows the test failure in issue #3579

File: server/src/main/java/org/opensearch/cluster/service/ClusterApplierService.java [hunk 1]
```diff
@@ -277,7 +277,7 @@ public void removeTimeoutListener(TimeoutClusterStateListener listener) {
     /**
      * Add a listener for on/off local node cluster-manager events
      */
-    public void addLocalNodeMasterListener(LocalNodeMasterListener listener) {
+    public void addLocalNodeMasterListener(LocalNodeClusterManagerListener listener) {
```
[the remark this answers is not in the document]
Comment (tlfeng) [reply server/src/main/java/org/opensearch/cluster/service/ClusterApplierService.java:280]: I will rename the method `addLocalNodeMasterListener` in issue #3544, which is being changed in PR https://github.com/opensearch-project/OpenSearch/pull/3647.
In issue #3543, only the methods within the above list of classes will be renamed, and of course, in a following PR as well.

File: test/framework/src/main/java/org/opensearch/test/ClusterServiceUtils.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ public static MasterService createMasterService(ThreadPool threadPool, ClusterSt
         return clusterManagerService;
     }
 
-    public static MasterService createMasterService(ThreadPool threadPool, DiscoveryNode localNode) {
+    public static ClusterManagerService createMasterService(ThreadPool threadPool, DiscoveryNode localNode) {
```
[the remark this answers is not in the document]
Comment (tlfeng) [reply test/framework/src/main/java/org/opensearch/test/ClusterServiceUtils.java:80]: Thanks! I will rename it when resolving issue #3544

File: server/src/main/java/org/opensearch/cluster/service/ClusterService.java [hunk 3]
```diff
@@ -214,12 +214,12 @@ public void removeListener(ClusterStateListener listener) {
...
      * Add a listener for on/off local node cluster-manager events
      */
-    public void addLocalNodeMasterListener(LocalNodeMasterListener listener) {
+    public void addLocalNodeMasterListener(LocalNodeClusterManagerListener listener) {
         clusterApplierService.addLocalNodeMasterListener(listener);
     }
 
-    public MasterService getMasterService() {
-        return masterService;
+    public ClusterManagerService getMasterService() {
```
[the remark this answers is not in the document]
Comment (tlfeng) [reply server/src/main/java/org/opensearch/cluster/service/ClusterService.java:221]: I will rename it when resolving issue #3544, which is being changed in PR https://github.com/opensearch-project/OpenSearch/pull/3647.

**Comment by tlfeng** [comment]
@Poojita-Raj Thanks a again for your approval! 👍👍
Please hold on backporting this change to 2.x branch, since I need to revise the plan for the renamed classes to make sure the compatibility is kept.

**Review by nknize — APPROVED** [review]
Looks like there's a merge conflict and a couple PR comments that still need to be addressed but otherwise LGTM.

**Pull request merged**